todolist/file_store.go

```diff
@@ -36,45 +47,68 @@ func (f *FileStore) Initialize() {
 
 func (f *FileStore) Load() ([]*Todo, error) {
 	if f.FileLocation == "" {
-		f.FileLocation = getLocation()
+		var err error
+		f.FileLocation, err = getLocation()
+		if err != nil {
+			fmt.Println("Error whlie finding todo file", err)
+		}
 	}
 
 	data, err := ioutil.ReadFile(f.FileLocation)
 	if err != nil {
 		fmt.Println("No todo file found!")
 		fmt.Println("Initialize a new todo repo by running 'todo init'")
 		os.Exit(0)
-		return nil, err
 	}
 
 	var todos []*Todo
 	jerr := json.Unmarshal(data, &todos)
 	if jerr != nil {
 		fmt.Println("Error reading json data", jerr)
 		os.Exit(1)
-		return nil, jerr
 	}
 	f.Loaded = true
 
 	return todos, nil
 }
 
 func (f *FileStore) Save(todos []*Todo) {
+	os.MkdirAll(filepath.Dir(f.FileLocation), 0700)
+
 	data, _ := json.Marshal(todos)
 	if err := ioutil.WriteFile(f.FileLocation, []byte(data), 0644); err != nil {
 		fmt.Println("Error writing json file", err)
 	}
 }
 
-func getLocation() string {
-	localrepo := ".todos.json"
-	usr, _ := user.Current()
-	homerepo := fmt.Sprintf("%s/.todos.json", usr.HomeDir)
-	_, ferr := os.Stat(localrepo)
+func getLocation() (location string, err error) {
+	// Look for `.todos.json` in current working directory
+	wd, err := os.Getwd()
+	if err != nil {
+		return
+	}
+	localrepo := filepath.Join(wd, ".todos.json")
+	if fileExists(localrepo) {
+		return localrepo, nil
+	}
 
-	if ferr == nil {
-		return localrepo
-	} else {
-		return homerepo
+	// Look for `.todos.json` in user's home directory
+	// (For compatability with old inits, see #72)
+	usr, err := user.Current()
+	if err != nil {
+		return
+	}
+	homerepo := filepath.Join(usr.HomeDir, ".todos.json")
+	if fileExists(homerepo) {
+		return homerepo, nil
 	}
+
+	// Default to the global todolist, whether it exists or not
+	globalrepo := filepath.Join(stddirs.DataDir(APP_ID), "todos.json")
+	return globalrepo, nil
+}
```

Comment:
I think it makes sense to split these out into separate functions for readability.  That way you wouldn't need the comments.

`return (localTodosFile || userHomeDirTodosFile || userDataDirTodosFile)`

...something like that.